hail/python/hailtop/batch_client/aioclient.py

```diff
@@ -56,17 +56,14 @@ def total_duration(job_status):
             return None
 
         durations = job_status['duration']
-        durations = [durations[task] for task in ['setup', 'main', 'cleanup'] if task in durations]
 
-        i = 0
-        duration = 0
-        while i < len(durations):
-            d = durations[i]
-            if d is None:
-                return None
-            duration += d
-            i += 1
-        return duration
+        setup_duration = durations.get('setup', 0)
+        main_duration = durations.get('main', 0)
+        cleanup_duration = durations.get('cleanup', 0)
+        if setup_duration is None or main_duration is None or cleanup_duration is None:
+            return None
+
+        return setup_duration + max(main_duration, cleanup_duration)
```

Comment:
I think this should be `setup_duration + main_duration + max(0, cleanup_duration - main_duration)`

Reply:
These are mathematically equivalent expressions and I think the current one is a bit simpler to read, no? Is there something I'm not thinking about?

Reply:
You're not accounting for the cleanup container running past the main container. i.e. copying files out.

Reply:
I see now.